includes/Discovery.php

```diff
@@ -262,22 +298,64 @@ public function print_twitter_metadata() {
 		if ( ! $enable_twitter_metadata ) {
 			return;
 		}
-		?>
-		<meta name="twitter:card" content="summary_large_image" />
-		<?php
+		$metadata = $this->get_twitter_metadata();
 
-		if ( ! has_post_thumbnail() ) {
-			return;
+		foreach ( $metadata as $name => $value ) {
+			printf( '<meta name="%s" content="%s" />', $name, $value );  // phpcs:ignore WordPress.Security.EscapeOutput.OutputNotEscaped
 		}
+	}
 
-		$poster = wp_get_attachment_image_url( (int) get_post_thumbnail_id(), Media::POSTER_PORTRAIT_IMAGE_SIZE );
+	/**
+	 * Get Twitter card metadata.
+	 *
+	 * @since 1.3.0
+	 *
+	 * @return array
+	 */
+	protected function get_twitter_metadata() {
+		$metadata = [
+			'twitter:card' => 'summary_large_image',
+		];
 
-		if ( ! $poster ) {
-			return;
+		/**
+		 * We're expecting a post object.
+		 *
+		 * @var WP_Post $post
+		 */
+		$post = get_queried_object();
+
+		if ( $post instanceof WP_Post ) {
+			if ( has_post_thumbnail( $post ) ) {
+				$poster_id = (int) get_post_thumbnail_id( $post );
+				if ( $poster_id ) {
+					$poster = wp_get_attachment_image_url( $poster_id, Media::POSTER_PORTRAIT_IMAGE_SIZE );
+
+					if ( $poster ) {
+						$metadata['twitter:image'] = esc_url( $poster );
+						$image_alt                 = get_post_meta( $poster_id, '_wp_attachment_image_alt', true );
```

Comment:
I doubt that the featured image will have good alt text, especially when it was uploaded through the editor, since its alt text cannot be edited there.

So I think we should — at least for now — just use the story title directly instead, and don't bother with the alt text.

Reply:
Fixed in https://github.com/google/web-stories-wp/pull/5919/commits/11ab8119541426348958a72ec412460d410e3c0d